backend/domain/domainmodels.js

```diff
@@ -0,0 +1,119 @@
+class Room {
+    constructor() {
+
+        this.roomName = this.setRoomName(); // some function to initialise a roomID
+
+        // default values given.
+        this.gamestate = GameStateEnum["Day Time"];
+
+        this.maxPlayerCount = 6;
+
+        this.players = new Array(maxPlayerCount);
+
+        this.roundNumber = 0;
+
+        this.getRoomName = function () {
+            return this.roomName;
+        };
+
+        this.getGamestate = function () {
+            return this.gamestate;
+        };
+
+        this.setGamestate = function (gamestate) {
+            this.gamestate = gamestate;
+        };
+
+        this.getPlayers = function () {
+            return this.players;
+        };
+
+        this.addPlayer = function (player) {
+            if (player !== null &&
+                this.getPlayers.length <= maxPlayerCount) {
+                this.players.push(player);
+            }
+        };
+
+        this.getRoundNumber = function () {
+            return this.roundNumber;
+        };
+
+        this.incrementRoundNumber = function () {
+            this.roundNumber++;
+        };
+
+        this.setRoomName = function () {
+            return 0; //some function to initialise a roomID. Must discuss how to do this at a later date.
+        };
+
+    }
+}
+
+class Player {
+    constructor(socketID, nickname, role) {
+
+        this.socketID = socketID;
+        this.nickname = nickname;
+        this.role = role;
+        this.isAlive = true;
+
+        this.getName = function () {
+            return this.name;
+        };
+
+        this.getNickname = function () {
+            return this.nickname;
+        };
+
+        this.getRole = function () {
+            return this.role;
+        };
+
+        this.getIsAlive = function () {
+            return this.isAlive;
+        };
+
+        this.setIsAlive = function () {
+            this.isAlive = !this.isAlive;
+        };
+
+    }
+} 
+
+class MafiaGame {
+    constructor() {
+
+        // Acts  as a mapping
+        const dict = {};
+        this.gameCount = 0;
+
+        this.newGame = function () {
+            incrementGameCoutner();
+            newRoom = new Room();
```

Comment:
`let newRoom = new Room();` . Don't we need a `let`. The same applies to `gameID` and other variables that are not in the constructor. It will bug out if we don't have a classifier

Reply:
you're right, Fixed